packages/astro/src/core/ssr/css.ts

```diff
@@ -1,34 +1,42 @@
 import type vite from '../vite';
 
 import path from 'path';
-import { viteifyURL } from '../util.js';
+import { viteID, viteURL } from '../util.js';
 
 // https://vitejs.dev/guide/features.html#css-pre-processors
 export const STYLE_EXTENSIONS = new Set(['.css', '.pcss', '.postcss', '.scss', '.sass', '.styl', '.stylus', '.less']);
 
-/** find unloaded styles */
+/**
+ * getStylesForURL
+ * given a URL, find all Vite ModuleGraph styles loaded
+ * Note: Vite has 2 lookup modes in the module graph: `id` and `url`. We have to search both to make sure we’re not missing any styles
+ */
 export function getStylesForURL(filePath: URL, viteServer: vite.ViteDevServer): Set<string> {
   const css = new Set<string>();
-  const rootID = viteifyURL(filePath);
 
   // recursively crawl module graph to get all style files imported by parent id
-  function crawlCSS(entryModule: string, scanned = new Set<string>()) {
-    const moduleName = viteServer.moduleGraph.urlToModuleMap.get(entryModule);
-    if (!moduleName || !moduleName.id) return;
+  function crawlCSS({ id, url, scanned = new Set<string>() }: { id: string | null; url: string; scanned?: Set<string> }) {
+    const moduleName = viteServer.moduleGraph.idToModuleMap.get(id || '') || viteServer.moduleGraph.urlToModuleMap.get(url);
```

Comment:
2/2 then the rest of the fix is just using both `id` and `url` to perform lookups. The order of `url` vs `id` doesn’t appear to be significant (i.e. it’s hard to say which is faster / more reliable). But we simply try both for a match, and use that to continue traversing Vite’s module graph.

It’s important to note that **the output is unchanged from this PR!** We end up with 100% the same output. We simply try more lookups to find that module if the first failed, which is the first time we’re doing this.